src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -3584,31 +3585,48 @@ def add_vnet_integration(cmd, name, resource_group_name, vnet, subnet, slot=None
               https://go.microsoft.com/fwlink/?linkid=2060115&clcid=0x409""")
 
     subnet_id_parts = parse_resource_id(subnet_resource_id)
+    subnet_subscription_id = subnet_id_parts['subscription']
     vnet_name = subnet_id_parts['name']
     vnet_resource_group = subnet_id_parts['resource_group']
     subnet_name = subnet_id_parts['child_name_1']
 
-    subnetObj = vnet_client.subnets.get(vnet_resource_group, vnet_name, subnet_name)
-    delegations = subnetObj.delegations
-    delegated = False
-    for d in delegations:
-        if d.service_name.lower() == "microsoft.web/serverfarms".lower():
-            delegated = True
+    if skip_delegation_check:
+        logger.warning('Skipping delegation check. Ensure that subnet is delegated to Microsoft.Web/serverFarms.'
+                       ' Missing delegation can cause "Bad Request" error.')
+    else:
```

Comment:
`else` means None or False, right? So when the value is None, you will also check delegation, right?

Reply:
Yes - delegation check should always be done, unless explicitly opting out.